src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -629,6 +637,27 @@ export function getInputType(
     }
 }
 
+export function navigateModels(
+    program: Program,
+    namespace: Namespace,
+    models: Map<string, InputModelType>,
+    enums: Map<string, InputEnumType>
+) {
+    const computeModel = (x: Type) => getInputType(program, x, models, enums);
+    const skipSubNamespaces = isGlobalNamespace(program, namespace);
+    navigateTypesInNamespace(
+        namespace,
+        {
+            model: (x) =>
+                x.name !== "" && x.kind === "Model" && computeModel(x),
+            scalar: computeModel,
+            enum: computeModel,
+            union: (x) => x.name !== undefined && computeModel(x)
+        },
+        { skipSubNamespaces }
+    );
```

Comment:
I believe this is the place that we are discussing in https://github.com/Azure/autorest.csharp/pull/2974/files#r1121103624
can we do this or similar to cast the return value of getInputType to any?
```suggestion
    const computeModel = (x: Type) => getInputType(program, x, models, enums) as any;
    const skipSubNamespaces = isGlobalNamespace(program, namespace);
    navigateTypesInNamespace(
        namespace,
        {
            model: (x) =>
                x.name !== "" && x.kind === "Model" && computeModel(x),
            scalar: computeModel,
            enum: computeModel,
            union: (x) => x.name !== undefined && computeModel(x)
        },
        { skipSubNamespaces }
    );
```
Or explicitly write the type of function `computeModel`? I do not know exactly what does the syntax looks like in typescript, only demo my point:
```suggestion
    const computeModel: Func<Type, any> = (x: Type) => getInputType(program, x, models, enums);
    const skipSubNamespaces = isGlobalNamespace(program, namespace);
    navigateTypesInNamespace(
        namespace,
        {
            model: (x) =>
                x.name !== "" && x.kind === "Model" && computeModel(x),
            scalar: computeModel,
            enum: computeModel,
            union: (x) => x.name !== undefined && computeModel(x)
        },
        { skipSubNamespaces }
    );
```

Reply:
Thanks @ArcturusZhang  @pshao25 , I use casting.